c7n/resources/aws.py

```diff
@@ -536,9 +575,22 @@ def __init__(self, ctx, config):
         super().__init__(ctx, config)
         # can't use a local session as we dont want an unassumed session cached.
         s3_client = self.ctx.session_factory(assume=False).client('s3')
-        region = s3_client.get_bucket_location(Bucket=self.bucket)['LocationConstraint']
-        # if region is None, we use us-east-1
-        region = region or "us-east-1"
+
+        # We don't yet use the S3 client to make an API call, but we do
+        # use it to find an appropriate regional endpoint for the active
+        # AWS partition
+        region = get_bucket_region_clientless(self.bucket, s3_client._endpoint.host)
+
+        if not region:
+            # If we can't determine a region without a client, try the
+            # GetBucketLocation API action.
+            region = s3_client.get_bucket_location(Bucket=self.bucket)['LocationConstraint']
+        if not region:
```

Comment:
region here could be None for us-east-1 fyi, it's probably better to do a try/except on the get_bucket_location call instead if you want to raise

Reply:
Hmm I was actually thinking about going the other way and ripping the get_bucket_location call out entirely. Between the unintuitive return values (None = 'us-east-1', EU = 'us-west-1) and access issues, HTTP requests seem more reliable. Left this in as a backstop though, because I assumed there was a case I was missing :).

Valid point though, if we're going to make the call we should be thorough and catch the "weird" regions...